services/web/src/screens/Auth/Login.js

```diff
@@ -1,28 +1,76 @@
-import React, { useState } from 'react';
+import { useState } from 'react';
+
+import {
+  Button,
+  Group,
+  Paper,
+  PasswordInput,
+  Stack,
+  TextInput,
+  Title,
+  Text,
+  Anchor,
+} from '@mantine/core';
+
+import { isEmail, useForm } from '@mantine/form';
+
 import { Link, useNavigate } from '@bedrockio/router';
-import { Form, Grid, Segment } from 'semantic';
 
 import { useSession } from 'stores/session';
 import Meta from 'components/Meta';
 
 import Federated from 'components/Auth/Federated';
 import ErrorMessage from 'components/ErrorMessage';
-import EmailField from 'components/form-fields/Email';
-import OptionalPassword from 'components/Auth/OptionalPassword';
-import Logo from 'components/LogoTitle';
+import Logo from 'components/Logo';
 
 import { request } from 'utils/api';
 import { AUTH_TYPE, AUTH_CHANNEL } from 'utils/env';
 
+function login(values) {
+  if (AUTH_TYPE === 'password') {
+    return loginPassword(values);
+  } else {
+    return loginOtp(values);
+  }
+}
+
+async function loginPassword(body) {
+  return await request({
+    method: 'POST',
+    path: `/1/auth/password/login`,
+    body,
+  });
+}
+
+async function loginOtp(body) {
+  return await request({
+    method: 'POST',
+    path: `/1/auth/otp/send`,
+    body: {
+      ...body,
+      type: AUTH_TYPE,
+      authChannel: AUTH_CHANNEL,
+    },
+  });
+}
+
 export default function PasswordLogin() {
   const navigate = useNavigate();
   const { authenticate } = useSession();
 
+  const form = useForm({
+    mode: 'uncontrolled',
```

Comment:
What is all this doing? I don't think I know of any use for an uncontrolled component... also what is `initialValues` here... can't this just be set in the state itself?